### PR TITLE
Kotlin Conversion: LanguageUtil

### PR DESCRIPTION
**Pull request by sharvaniharan**

### CODE REVIEW

File: app/src/main/java/org/wikipedia/language/LanguageUtil.kt [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import org.apache.commons.lang3.StringUtils
+import org.wikipedia.WikipediaApp
+import org.wikipedia.util.StringUtil
+import java.util.*
+
+object LanguageUtil {
+
+    private const val HONG_KONG_COUNTRY_CODE = "HK"
+    private const val MACAU_COUNTRY_CODE = "MO"
+    private val TRADITIONAL_CHINESE_COUNTRY_CODES = mutableListOf(Locale.TAIWAN.country, HONG_KONG_COUNTRY_CODE, MACAU_COUNTRY_CODE)
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/language/LanguageUtil.kt:16]: The list should be non-mutable and can use `listOf()`.

File: app/src/main/java/org/wikipedia/language/LanguageUtil.kt [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    @JvmStatic
+    val firstSelectedChineseVariant: String
+        get() {
+            var firstSelectedChineseLangCode: String? = null
+            for (langCode in WikipediaApp.getInstance().language().appLanguageCodes) {
+                if (langCode.startsWith(AppLanguageLookUpTable.CHINESE_LANGUAGE_CODE)) {
+                    firstSelectedChineseLangCode = langCode
+                    break
+                }
+            }
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/language/LanguageUtil.kt:125]: Can we use `WikipediaApp.getInstance().language().appLanguageCodes.firstOrNull(...)` here?

File: app/src/main/java/org/wikipedia/language/LanguageUtil.kt [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            return StringUtils.defaultString(firstSelectedChineseLangCode,
+                AppLanguageLookUpTable.TRADITIONAL_CHINESE_LANGUAGE_CODE)
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/language/LanguageUtil.kt:127]: Maybe use `firstSelectedChineseLangCode.orEmpty().ifEmpty {  }`?

File: app/src/main/java/org/wikipedia/language/LanguageUtil.kt [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        }
+
+    fun startsWithArticle(text: String, language: String): Boolean {
+        val first = text.split(" ".toRegex()).toTypedArray()[0].lowercase(Locale.getDefault()).trim { it <= ' ' }
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/language/LanguageUtil.kt:131]: We may just use `.trim()` instead of `.trim { it <= ' ' }`

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/language/LanguageUtil.kt (hunk 3 above)
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/language/LanguageUtil.kt:127]: isnt this cleaner?

**Pull request merged**